From b9cb30c7fe0b1fceaa10bae75b94c482c2699645 Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Thu, 10 Apr 2014 22:36:01 -0400
Subject: [PATCH] lib/keyboard.py: make sure etc/sysconfig/ exists

Make sure etc/sysconfig exists before write to etc/sysconfig/keyboard
since it may not exist when we do the image installation based
kickstart.

Upstream-Status: Pending

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 lib/keyboard.py |    7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/lib/keyboard.py b/lib/keyboard.py
index 5b0ac00..a8c8b91 100644
--- a/lib/keyboard.py
+++ b/lib/keyboard.py
@@ -133,11 +133,14 @@ class Keyboard():
         self.beenset = 1
 
     def write(self, instPath = "/"):
-	file = open(instPath + "/etc/sysconfig/keyboard", "w")
+	sysconfig_dir = os.path.join(instPath + "/etc/sysconfig")
+        if not os.path.exists(sysconfig_dir):
+                os.makedirs(sysconfig_dir)
+	file = open(sysconfig_dir + "/keyboard", "w")
 	for line in self.config:
                 file.write (line[0]);
         try:
-            os.unlink(instPath + "/etc/sysconfig/console/default.kmap")
+            os.unlink(sysconfig_dir + "/console/default.kmap")
         except:
             pass
 
-- 
1.7.10.4

